File: 8-kyu/sum-of-differences-in-array/c++/solution.cpp
#include <algorithm>
#include <numeric>
#include <vector>

int sumOfDifferences(const std::vector<int>& arr)
{
    if (arr.size() < 2) return 0;
    auto arr_sorted = arr;
    int ret = 0;
    std::sort(arr_sorted.begin(), arr_sorted.end(), std::greater<int>());
    for (unsigned i = 0; i < arr_sorted.size() - 1; i++) {
        ret += arr_sorted[i] - arr_sorted[i + 1];
    }
    return ret;
}

File: 8-kyu/sum-of-differences-in-array/javascript/solution.js
const sumOfDifferences = (arr) => {
  if (arr.length < 2) return 0;
  let arr2 = arr.sort((a, b) => b - a);
  let ret = 0;
  for (let i = 0; i < arr.length - 1; i++) {
    ret += arr[i] - arr[i + 1];
  }
  return ret;
};